lib/spack/spack/binary_distribution.py

```diff
@@ -2199,11 +2204,26 @@ def relocate_package(spec):
     # First match specific prefix paths. Possibly the *local* install prefix
     # of some dependency is in an upstream, so we cannot assume the original
     # spack store root can be mapped uniformly to the new spack store root.
-    for dag_hash, new_dep_prefix in hashes_to_prefixes(spec).items():
-        if dag_hash in hash_to_old_prefix:
-            old_dep_prefix = hash_to_old_prefix[dag_hash]
-            prefix_to_prefix_bin[old_dep_prefix] = new_dep_prefix
-            prefix_to_prefix_text[old_dep_prefix] = new_dep_prefix
+    for dep in deps_to_relocate(spec):
+        try:
+            lookup_dag_hash = spec.build_spec[dep.name].dag_hash()
```

Comment:
Due to reuse you can have multiple instances of `gcc-runtime` among link deps. So using `spec[name]` is a mistake: the current implementation produces a single mapping `<old gcc-runtime prefix>` => <new gcc-runtime prefix>` instead of relocation all instances.